lib/ccf/index.ts

```diff
@@ -0,0 +1,378 @@
+                'vCPUs': cpus,
+                'consumption': {
+                    'minWatts': this.gcpList[architecture]['Min Watts'] * cpus,
+                    'maxWatts': this.gcpList[architecture]['Max Watts'] * cpus,
+                },
+                'maxvCPUs': parseInt(instance['Platform vCPUs (highest vCPU possible)'], 10)
+            } as IComputeInstance;
+        });
+        azure_instances.forEach((instance: { [key: string]: any }) => {
+            const cpus = parseInt(instance['Instance vCPUs'], 10);
+            let architecture = instance['Microarchitecture'];
+            if (!(architecture in this.azureList)) {
+                architecture = 'Average';
+            }
+            this.computeInstances['azure'][instance['Virtual Machine']] = {
+                consumption: {
+                    'minWatts': this.azureList[architecture]['Min Watts'] * cpus,
+                    'maxWatts': this.azureList[architecture]['Max Watts'] * cpus,
+                },
+                'name': instance['Virtual Machine'],
+                'vCPUs': instance['Instance vCPUs'],
+                'maxvCPUs': parseInt(instance['Platform vCPUs (highest vCPU possible)'], 10)
+            } as IComputeInstance;
+        });
+        aws_embodied.forEach((instance: { [key: string]: any }) => {
+            this.computeInstances['aws'][instance['type']].embodiedEmission = instance['total'];
+        });
+        gcp_embodied.forEach((instance: { [key: string]: any }) => {
+            this.computeInstances['gcp'][instance['type']].embodiedEmission = instance['total'];
+        });
+        azure_embodied.forEach((instance: { [key: string]: any }) => {
+            this.computeInstances['azure'][instance['type']].embodiedEmission = instance['total'];
+        });
+    }
+
+    // Architecture strings are different between Instances-Use.JSON and the bundled Typescript from CCF.
+    // This function resolves the differences.
+    private resolveAwsArchitecture(architecture: string) {
+        if (architecture.includes('AMD ')) {
+            architecture = architecture.substring(4);
+        }
+        if (architecture.includes("Skylake")) {
+            architecture = "Sky Lake";
+        }
+        if (architecture.includes('Graviton')) {
+            if (architecture.includes('2')) {
+                architecture = 'Graviton2';
+            } else if (architecture.includes('3')) {
+                architecture = 'Graviton3';
+            } else {
+                architecture = 'Graviton';
+            }
+        }
+        if (architecture.includes('Unknown')) {
+            architecture = 'Average';
+        }
+        if (!(architecture in this.awsList)) {
+            console.log("ARCHITECTURE:", architecture)
+        }
+        return architecture;
+    }
+
+    // Calculates the embodied emissions for a given observation
+    embodiedEmissions(observation: { [key: string]: any; }): number {
```

Comment:
in my PR I have introduced types for `{ [key: string]: any; }` this kind of things, so after merge we can reuse them in this code too.